packages/hoppscotch-common/src/services/workspace.service.ts

```diff
@@ -101,6 +101,61 @@ export class WorkspaceService extends Service<WorkspaceServiceEvent> {
       },
       { immediate: true }
     )
+
+    // Watch for workspace changes and update team collection service accordingly
+    this.setupTeamCollectionServiceSync()
+  }
+
+  /**
+   * Sets up synchronization with team collection service
+   * This ensures team collections are updated when workspace changes
+   */
+  private setupTeamCollectionServiceSync() {
+    watch(
+      this._currentWorkspace,
+      async (newWorkspace, oldWorkspace) => {
+        // Skip update if workspaces are effectively the same
+        if (this.areWorkspacesEqual(newWorkspace, oldWorkspace)) return
+
+        try {
+          // Use dynamic import to avoid circular dependency
+          const { TeamCollectionsService } = await import(
+            "./team-collection.service"
+          )
+          const teamCollectionService = this.bind(TeamCollectionsService)
+
+          if (newWorkspace.type === "team" && newWorkspace.teamID) {
+            teamCollectionService.changeTeamID(newWorkspace.teamID)
+          } else {
+            teamCollectionService.clearCollections()
+          }
+        } catch (error) {
+          console.error("Failed to sync team collections:", error)
+        }
+      },
+      { immediate: true }
+    )
+  }
+
+  /**
+   * Checks if two workspaces are effectively equal to avoid unnecessary updates
+   */
+  private areWorkspacesEqual(
+    newWorkspace: Workspace | undefined,
+    oldWorkspace: Workspace | undefined
```

Comment:
Is there a scenario where `newWorkspace` can be `undefined`? The inferred type contracts from the watcher don't indicate. However, there's the runtime nuance with `oldWorkspace` resolved as `undefined` during the first load.